### PR TITLE
Adding first version of Edge gateway sdk and tests

### PR DESCRIPTION
**Pull request by anponnet**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by anponnet** [comment]
Swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/5191/commits

**Comment by anponnet** [comment]
swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/5191/commits

**Comment by dsgouda** [comment]
@anponnet please confirm that you have joined the Azure org [here](aka.ms/azuregithub)

File: src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj
Comment (dsgouda) [inline src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj]: Please rename this to `EdgeGateway.Tests`

File: src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj
Comment (dsgouda) [inline src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj]: Please replace this with `<Version>1.0.0</Version>`

File: src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj
Comment (dsgouda) [inline src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj]: Please update the `AssemblyName`

File: src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj
Comment (dsgouda) [inline src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj]: Incorrect name

Comment (dsgouda) [inline src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj]: Please add the line
`<TreatWarningsAsErrors>true</TreatWarningsAsErrors>`
And fix errors if any

File: src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj
Comment (dsgouda) [inline src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj]: We recommend starting versioning from `0.9.0-preview` before releasing a stable version. Please confirm whether you are ready to release a stable version of the SDK?

File: src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj
Comment (dsgouda) [inline src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj]: Any reason why this was added?

**Review by dsgouda — CHANGES_REQUESTED** [review]
How was this code generated? Please create a generate.ps1 file similar to [this](https://github.com/Azure/azure-sdk-for-net/blob/master/src/SDKs/Graph.RBAC/Graph.RBAC/generate.ps1) and regenerate the code. Please commit all the changes generated.

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please add `AssemblyInfo.cs` files for Test and Sdk projects.

File: src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj
Comment (anponnet) [reply src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj]: Hi @dsgouda 
You have asked to use generate.ps1 (this) to generated client code. But from where it will pick swagger? I think it is picking from https://github.com/Azure/azure-rest-api-specs. We do not have our latest swagger there yet (It is in PR https://github.com/Azure/azure-rest-api-specs/pull/5191). 

I had used below command to generate client code. The client code I get from both are different, is there I way I can specify swagger file to use while generating with the method you mentioned

autorest -CodeGenerator Azure.CSharp -Modeler Swagger -Input "C:\Users\anponnet\Desktop\Petstore\swagger.json" -Namespace Microsoft.Azure.Management.EdgeGateway

**Comment by anponnet** [comment]
@dsgouda I am part of azure org

File: src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj
Comment (anponnet) [reply src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj]: Fixed

File: src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj
Comment (anponnet) [reply src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj]: Fixed

File: src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj
Comment (anponnet) [reply src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj]: Fixed

File: src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj
Comment (anponnet) [reply src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj]: Fixed

Comment (anponnet) [reply src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj]: Fixed

File: src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj
Comment (anponnet) [reply src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj]: Changed to 0.9.0-preview. May be we will release first version after 1-2 weeks

File: src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj
Comment (anponnet) [reply src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj]: Removed the line

File: src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj
Comment (anponnet) [reply src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj]: Corrected using generate.ps1. Swagger was taken from the fork. https://github.com/anponnet/azure-rest-api-specs/tree/version20190301

File: src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj
Comment (dsgouda) [inline src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj]: Please replace this with ` <TargetFrameworks>$(SdkTargetFx)</TargetFrameworks>`

File: src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj
Comment (dsgouda) [inline src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj]: Don't think this line is needed

File: src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj
Comment (dsgouda) [inline src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj]: Please remove this `ItemGroup`

File: src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Properties/AssemblyInfo.cs
Comment (dsgouda) [inline src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Properties/AssemblyInfo.cs]: `AssemblyVersion` must also be `0.9.0.0`

File: src/SDKs/StorSimple/Series1200/Management.StorSimple1200Series/generate.ps1
Comment (dsgouda) [inline src/SDKs/StorSimple/Series1200/Management.StorSimple1200Series/generate.ps1]: Please undo changes to this file

File: src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj
Comment (dsgouda) [inline src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj]: Remove lines 16 and 17

File: src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj
Comment (dsgouda) [inline src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj]: Please use a regex like `SessionRecords\**\*.json` instead

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Properties/AssemblyInfo.cs
Comment (dsgouda) [inline src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Properties/AssemblyInfo.cs]: Please fix `AssemblyFileVersion` too.

File: src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj
Comment (anponnet) [reply src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj]: fixed

File: src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Properties/AssemblyInfo.cs
Comment (dsgouda) [inline src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Properties/AssemblyInfo.cs]: I probably misspoke earlier, we start versioning for new packages from `0.8.0-preview`
Please update the csproj and this file to set all versions to `0.8.0` (-preview in the csproj)

File: src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj
Comment (anponnet) [reply src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj]: fixed

File: src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Properties/AssemblyInfo.cs
Comment (anponnet) [reply src/SDKs/EdgeGateway/Microsoft.Azure.Management.EdgeGateway/Properties/AssemblyInfo.cs]: fixed

File: src/SDKs/StorSimple/Series1200/Management.StorSimple1200Series/generate.ps1
Comment (anponnet) [reply src/SDKs/StorSimple/Series1200/Management.StorSimple1200Series/generate.ps1]: fixed

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks great apart from a couple of minor corrections

File: src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj
Comment (anponnet) [reply src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj]: fixed

File: src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj
Comment (anponnet) [reply src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj]: fixed

**Review by dsgouda — CHANGES_REQUESTED** [review]
Also, please rename the directory `Microsoft.Azure.Management.EdgeGateway` to `Management.EdgeGateway` for consistency

**Review by dsgouda — COMMENTED** [review]
Approving in principle, will merge once Rest spec is merged and code is regenerated

**Review by dsgouda — COMMENTED** [review]
Please remove the `$(SdkTargetFx)` file

**Comment by anponnet** [comment]
> Please remove the $(SdkTargetFx) file

@dsgouda I did not understand the comment. Should I remove generate.ps1 file?

**Comment by anponnet** [comment]
@dsgouda :  Rest spec is merged and code is regenerated

**Comment by dsgouda** [comment]
@anponnet can you check if you have joined the Azure org [here](http://aka.ms/azuregithub)

**Comment by dsgouda** [comment]
@azuresdkci Test this please

**Comment by dsgouda** [comment]
@anponnet please note that we have switched our base branch to master. That was the problem here.

File: src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj
Comment (dsgouda) [inline src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj]: Please remove these conditions

File: src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj
Comment (dsgouda) [inline src/SDKs/EdgeGateway/EdgeGateway.Tests/EdgeGateway.Tests.csproj]: Please replace `VersionPrefix` with `Version`

File: src/SDKs/EdgeGateway/Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj
Comment (dsgouda) [inline src/SDKs/EdgeGateway/Management.EdgeGateway/Microsoft.Azure.Management.EdgeGateway.csproj]: nit: "public preview release"

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please delete the file  `$(SdkTargetFx)`

**Review by dsgouda — APPROVED** [review]
LGTM will merge once CIs pass

**Pull request merged**
